source/geometry/FloatConvexHull2D.ooc

```diff
@@ -102,24 +97,26 @@ FloatConvexHull2D: class {
 		if (currentSet count == 1)
 			this _points add(currentSet[0])
 		else if (currentSet count > 1) {
-			maximumDistance := Float negativeInfinity
-			maximumDistanceIndex := -1
+			(maximumDistance, maximumDistanceIndex) := (Float negativeInfinity, -1)
 			for (i in 0 .. currentSet count) {
-				distance := This _pointsLinePseudoDistance(leftPoint, rightPoint, currentSet[i])
-				if (distance > maximumDistance) {
-					maximumDistance = distance
-					maximumDistanceIndex = i
-				}
+				queryPoint := currentSet[i]
+				pseudoDistance := ((rightPoint y - leftPoint y) * (leftPoint x - queryPoint x)) - ((rightPoint x - leftPoint x) * (leftPoint y - queryPoint y))
+				if (pseudoDistance > maximumDistance)
+					(maximumDistance, maximumDistanceIndex) = (pseudoDistance, i)
 			}
 			maximumDistancePoint := currentSet[maximumDistanceIndex]
-			outsideLeft := VectorList<FloatPoint2D> new()
-			outsideRight := VectorList<FloatPoint2D> new()
-			for (i in 0 .. currentSet count)
-				if (i != maximumDistanceIndex)
-					if (This _isOnLeft(leftPoint, maximumDistancePoint, currentSet[i]))
-						outsideLeft add(currentSet[i])
-					else if (This _isOnLeft(maximumDistancePoint, rightPoint, currentSet[i]))
-						outsideRight add(currentSet[i])
+			outsideLeft := VectorList<FloatPoint2D> new(currentSet _count)
+			outsideRight := VectorList<FloatPoint2D> new(currentSet _count)
+			for (i in 0 .. maximumDistanceIndex)
```

Comment:
Have you tested that the compiler does not actually do that with -O3 for you ? I think previous loop code was more readable